### PR TITLE
Replace CircleCI with GitHub actions

### PR DESCRIPTION
**Pull request by thomaseizinger**
Work towards #105. 

With this PR, we do a debug build for each platform on every build. GitHub actions also allows us to download the artifact.

Once the cache limit for GitHub actions is increased OR we strip down on some of our dependencies, we might also be able to use to the cache action to further speed up the build.

### CODE REVIEW

File: src/env/clean_up.rs [hunk 1]
```diff
@@ -24,7 +24,12 @@ pub fn register_signals() -> anyhow::Result<Arc<AtomicBool>> {
     let terminate = Arc::new(AtomicBool::new(false));
     signal_hook::flag::register(signal_hook::SIGTERM, Arc::clone(&terminate))?;
     signal_hook::flag::register(signal_hook::SIGINT, Arc::clone(&terminate))?;
-    signal_hook::flag::register(signal_hook::SIGQUIT, Arc::clone(&terminate))?;
+
+    #[cfg(not(windows))]
```
Comment (luckysori) [inline src/env/clean_up.rs]: :question: Will clean-up work the same on Windows then?

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        uses: actions/upload-artifact@v1
+        with:
+          name: create-comit-app-${{ matrix.os }}-debug
+          path: target/debug/${{ matrix.binary }}
+
+  e2e_test:
+    strategy:
+      matrix:
+        os: [ ubuntu ]
+    runs-on: ${{ matrix.os }}-latest
```
Comment (D4nte) [inline .github/workflows/CI.yml]:
```suggestion
    runs-on: ubuntu-latest
```

File: tests/new.sh [hunk 3]
```diff
@@ -7,7 +7,7 @@ PROJECT_DIR=${0%/tests/*.sh}
 CCA="${PROJECT_DIR}/target/debug/create-comit-app"
 
 # Random 10 char name
```
Comment (luckysori) [inline tests/new.sh]:
```suggestion
```

File: .mergify.yml [hunk 4]
```diff
@@ -1,8 +1,11 @@
 pull_request_rules:
   - name: instruct bors to merge dependabot PRs with passing tests
     conditions:
-      - "status-success=ci/circleci: build_test"
-      - "status-success=ci/circleci: static_analysis"
+      - "status-success=CI / static_analysis (push)"
```
Comment (D4nte) [inline .mergify.yml]: 📣 
```suggestion
      - "status-success=static_analysis"
```
This is how it should be named.
See comit-sdk config as proof/example.

File: .github/workflows/CI.yml [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+      - name: Upload ${{ matrix.os }} binary
+        uses: actions/upload-artifact@v1
+        with:
+          name: create-comit-app-${{ matrix.os }}-debug
+          path: target/debug/${{ matrix.binary }}
+
+  e2e_test:
+    strategy:
+      matrix:
```
Comment (luckysori) [inline .github/workflows/CI.yml]: :wrench: I guess you don't need a matrix if you only run it on 1 :stuck_out_tongue:

File: src/env/clean_up.rs (hunk 1 above)
Comment (D4nte) [reply src/env/clean_up.rs]: Yes should be fine as `CTRL+C` is `SIGINT` on windows: https://docs.microsoft.com/en-us/windows/console/ctrl-c-and-ctrl-break-signals

**Pull request merged**